packages/utilities/src/utilities/StickyHeader/StickyHeader.js

```diff
@@ -171,116 +186,207 @@ class StickyHeader {
+    const tocIsStuck =
+      Math.round(tableOfContentsInnerBar.getBoundingClientRect().top) <=
+      this._state.cumulativeOffset + 1;
 
-      if (masthead && tocIsAtTop && (tocShouldStick || mastheadL1)) {
-        maxScrollaway += masthead.offsetHeight;
+    if (tocShouldStick && tocIsStuck) {
+      this._state.cumulativeOffset += tableOfContentsInnerBar.offsetHeight;
+    }
+  }
 
-        if (mastheadL1 && !tocShouldStick) {
-          maxScrollaway -= mastheadL1.offsetHeight;
-        }
-      } else if (mastheadL0 && mastheadL1) {
-        maxScrollaway += mastheadL0.offsetHeight;
+  /**
+   * Handles the leadspace search given the current scroll position.
+   */
+  _handleLeadspaceSearch() {
+    const { leadspaceSearch, leadspaceSearchBar, leadspaceSearchInput } =
+      this._elements;
+    const { leadspaceSearchThreshold } = this._state;
+    const searchShouldBeSticky =
+      leadspaceSearch.getBoundingClientRect().bottom <=
+      leadspaceSearchThreshold;
+    const searchIsSticky = leadspaceSearch.hasAttribute('sticky-search');
+
+    if (searchShouldBeSticky) {
+      if (!searchIsSticky) {
+        leadspaceSearch.style.paddingBottom = `${leadspaceSearchBar.offsetHeight}px`;
+        leadspaceSearch.setAttribute('sticky-search', '');
+        leadspaceSearchInput.setAttribute('large', '');
+
+        window.requestAnimationFrame(() => {
+          leadspaceSearchBar.style.transitionDuration = '110ms';
+          leadspaceSearchBar.style.transform = 'translateY(0)';
+        });
       }
+      leadspaceSearchBar.style.top = `${this._state.cumulativeOffset}px`;
+      this._state.cumulativeOffset += leadspaceSearchBar.offsetHeight;
+    } else if (searchIsSticky) {
+      leadspaceSearch.style.paddingBottom = '';
+      leadspaceSearch.removeAttribute('sticky-search');
+      leadspaceSearchInput.removeAttribute('large');
+
+      leadspaceSearchBar.style.transitionDuration = '';
+      leadspaceSearchBar.style.transform = '';
+      leadspaceSearchBar.style.top = '';
     }
+  }
 
-    // Calculate maxScrollaway values based on leadspace search position
-    if (!tocInner && leadspaceSearchBar) {
-      const searchIsAtTop =
-        leadspaceSearchBar.getBoundingClientRect().top <=
-        (masthead ? masthead.offsetTop + masthead.offsetHeight : 0) + 1;
+  /**
+   * Calculates a value matching the height of all components that are allowed
+   * to hide above the viewport.
+   *
+   * Adding an item's height to this value indicates we expect it to be hidden
+   * above the viewport.
+   *
+   * Items that stick, in order
+   * - L0
+   * - L1
+   * - The TOC in horizontal bar form
+   * - The leadspace with search (if no TOC)
+   */
+  _calculateMaxScrollaway() {
+    const {
+      masthead,
+      mastheadL0,
+      mastheadL1,
+      tableOfContents,
+      tableOfContentsInnerBar,
+      leadspaceSearchBar,
+    } = StickyHeader.global._elements;
```

Comment:
Same note here as in `_manageStickyElements`. Since this method is called exclusively from `_manageStickyElements`, should be safe by the same rationale to use `this`.